src/github/taskDashboardWebviewProvider.ts

```diff
@@ -0,0 +1,1180 @@
+				return;
+			}
+
+			// Get the default branch (usually main or master)
+			const defaultBranch = await this.getDefaultBranch(folderManager) || 'main';
+
+			// Switch to the default branch
+			await folderManager.repository.checkout(defaultBranch);
+			vscode.window.showInformationMessage(`Switched to branch: ${defaultBranch}`);
+
+			// Update dashboard to reflect the branch change
+			setTimeout(() => {
+				this.updateDashboard();
+			}, 500);
+		} catch (error) {
+			Logger.error(`Failed to switch to main branch: ${error}`, DashboardWebviewProvider.ID);
+			vscode.window.showErrorMessage(`Failed to switch to main branch: ${error}`);
+		}
+	}
+
+	private async createPullRequest(): Promise<void> {
+		try {
+			// Find the first available folder manager with a repository
+			const folderManager = this._repositoriesManager.folderManagers.find(fm =>
+				fm.gitHubRepositories.length > 0
+			);
+
+			if (!folderManager) {
+				vscode.window.showErrorMessage('No GitHub repository found in the current workspace.');
+				return;
+			}
+
+			const repository = folderManager.repository;
+			const currentBranch = repository.state.HEAD?.name;
+
+			if (!currentBranch) {
+				vscode.window.showErrorMessage('No current branch found.');
+				return;
+			}
+
+			// Check if there are any commits on this branch that aren't on main
+			const hasCommits = await this.hasCommitsOnBranch(repository, currentBranch);
+
+			if (!hasCommits) {
+				// No commits yet, stage files, generate commit message, and open SCM view
+				try {
+					// Stage all changed files
+					const workingTreeChanges = repository.state.workingTreeChanges;
+					if (workingTreeChanges.length > 0) {
+						await repository.add(workingTreeChanges.map(change => change.uri.fsPath));
+						Logger.debug(`Staged ${workingTreeChanges.length} files`, DashboardWebviewProvider.ID);
+					}
+
+					// Open SCM view first
+					await vscode.commands.executeCommand('workbench.view.scm');
+
+					// Generate commit message using Copilot
+					try {
+						await vscode.commands.executeCommand('github.copilot.git.generateCommitMessage');
+					} catch (commitMsgError) {
+						Logger.debug(`Failed to generate commit message: ${commitMsgError}`, DashboardWebviewProvider.ID);
+						// Don't fail the whole operation if commit message generation fails
+					}
+
+					vscode.window.showInformationMessage('Files staged and commit message generated. Make your first commit before creating a pull request.');
+				} catch (stagingError) {
+					Logger.error(`Failed to stage files: ${stagingError}`, DashboardWebviewProvider.ID);
+					// Fall back to just opening SCM view
+					await vscode.commands.executeCommand('workbench.view.scm');
+					vscode.window.showInformationMessage('Make your first commit before creating a pull request.');
+				}
+			} else {
+				// Has commits, proceed with create pull request flow
+				await vscode.commands.executeCommand('pr.create');
```

Comment:
I have tried to remove most of the direct calls of PR extension commands like this, as it feels like code smell. You can call this create method:

https://github.com/microsoft/vscode-pull-request-github/blob/e5ce10696b0010b836f8a9a96a4ecd6a33295a48/src/view/reviewManager.ts#L1159-L1160